modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AbstractValidator.java

```diff
@@ -128,7 +140,40 @@ private static void validateSCIMObjectForRequiredSubAttributes(AbstractAttribute
                     }
                     List<AttributeSchema> subSubAttributesSchemaList = subAttributeSchema.getSubAttributeSchemas();
                     if (subSubAttributesSchemaList != null) {
-                        validateSCIMObjectForRequiredSubAttributes(subAttribute, subAttributeSchema);
+                        validateSCIMObjectForRequiredSubAttributes(subAttribute, subAttributeSchema, scimObject);
+                    }
+                }
+            }
+        }
+    }
+
+    /*
+     * Validate SCIMObject for canonical attributes in groups.
+     *
+     * @param attribute
+     * @param subAttributeSchema
+     * @param scimObject
+     * @param canonicalValues
+     * @throws CharonException
+     * @throws BadRequestException
```

Comment:
If we are adding a method comment, make sure to complete it with appropriate descriptions.